GetStream/stream-chat-android
# [13] Create internal logging tool

### 🎯 Goal

Ticket: https://github.com/GetStream/android-internal-board/issues/13

Introduce `StreamLog` - a new parameterized file logging.

### 🛠 Implementation details

The logging is getting enabled by default in debug builds only, cause `stream-logger-android` module is getting linked as a dependency using `debugApi`.

`stream-logger-android` module contains `StreamLogFileProvider` class which is getting invoked right before `Application.onCreate()`. This class initializes the `StreamLog` with android & file logging.

### 🧪 Testing

- Cut new debug build of any sample app.
- Launch app 
- send in terminal `adb shell am startservice -a io.getstream.logging.android.CLEAR`
- you should see the toast message `Logs cleared`
- open any channel and send a few messages
- send in terminal `adb shell am startservice -a io.getstream.logging.android.SHARE`
- you should see a file-sharing dialog chooser.
- share log file
- check it's not empty & contains log content.


#### The way to use `StreamLog`
##### Option 1:
```kotlin
private const val TAG = "SomeViewModel"
// ...
StreamLog.d(TAG) { "[<method_name>] arg1: $arg1, arg2: $arg2, arg3: $arg3" }
```
##### Option 2:
```kotlin
val logger = StreamLog.getLogger("SomeViewModel")
// ...
logger.d { "[<method_name>] arg1: $arg1, arg2: $arg2, arg3: $arg3" }
```

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_
